### PR TITLE
TLOZ: APworld support

### PR DESCRIPTION
**Pull request by t3hf1gm3nt**
## What is this fixing or adding?
- Remove a relative import in Rules.py
- Clean up a few unused imports in __init__.py
- Use pkgutil instead of open when applying base patch
- make sure rom_name is initialized correctly in modify_multidata

## How was this tested?
Created tloz.apworld with modified files, put in release installation of AP, generated game, tested that base patch was still applied by seeing that recorder went to all dungeons and that all dungeons already had map and compass.

## If this makes graphical changes, please attach screenshots.
N/A

### CODE REVIEW

**Comment by t3hf1gm3nt** [comment]
Also apologies for messy commit history. Unsure what happened there.

File: worlds/tloz/__init__.py [hunk 1]
```diff
@@ -168,9 +167,8 @@ def apply_base_patch(self, rom):
         # Remove map/compass check so they're always on
         # Removing a bit from the boss roars flags, so we can have more dungeon items. This allows us to
         # go past 0x1F items for dungeon items.
-        base_patch_location = os.path.dirname(__file__) + "/z1_base_patch.bsdiff4"
-        with open(base_patch_location, "rb") as base_patch:
-            rom_data = bsdiff4.patch(rom.read(), base_patch.read())
+        base_patch = get_data(__name__, os.path.dirname(__file__) + "/z1_base_patch.bsdiff4")
```
Comment (Berserker66) [inline worlds/tloz/__init__.py:170]: os.path.join() should be used over an explicit "/"

Comment (t3hf1gm3nt) [reply worlds/tloz/__init__.py:170]: So I've pushed a commit to address but github's being a butt atm and not updating the PR. hopefully it shows up SOONtm

Comment (t3hf1gm3nt) [reply worlds/tloz/__init__.py:170]: ....and 14 hours later it still isn't showing up. unsure what to do about this

Comment (t3hf1gm3nt) [reply worlds/tloz/__init__.py]: ok FINALLY got it to show up after merging in the newest main now that there was a new commit

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**